espressostreamer/espresso_streamer.go

```diff
@@ -124,8 +128,13 @@ func (s *EspressoStreamer) Reset(currentMessagePos uint64, currentHostshotBlock
 	s.messageLock.Lock()
 	defer s.messageLock.Unlock()
 
+	hotshotBlockNum := currentHostshotBlock
```

Comment:
I think this could make the batcher stuck if it ever needs to reset behind the configured minimum here, but given this is a dangerous config, I think it's fine.

I think this should just be removed when we implement fetching txns from a range of hotshot blocks in the `go-sdk`, and utilize that functionality in the `espresso-streamer`.  I'm more than happy to work on this with @ImJeremyHe at some point?